### ISSUE TITLE
No plugin display

### ISSUE THREAD

**Issue by sabas1080** [VedantParanjape/KiBuzzard#2]
Hi @VedantParanjape 

Thanks for your work. I am testing this in Kicad 5.1.9 and MacOS

but I have the next error

```
   from . import KiBuzzard
  File "/Library/Preferences/kicad/scripting/plugins/KiBuzzard/KiBuzzard/__init__.py", line 15, in <module>
    from .dialog.dialog import Dialog
ImportError: No module named dialog.dialog
```
I am tred with pip install pythondialog but same error

**Comment by VedantParanjape** [comment]
Are you sure you installed the requirements correctly and does the original port run on your system ?

**Comment by sabas1080** [comment]
Yes with pip install -r requirements.txt --user
<img width="883" alt="Captura de Pantalla 2021-01-17 a la(s) 14 38 52" src="https://user-images.githubusercontent.com/2874292/104855330-c2877b00-58d1-11eb-802f-300dae67f53f.png">

**Comment by VedantParanjape** [comment]
Where did this error show up ? 

Can you share output of the your .kicad/scripting/plugins directory. I can't reproduce this error. I think you haven't correctly installed the plugin

**Comment by sabas1080** [comment]
The error is in python shell Kicad

```
import KiBuzzard
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/Users/sabas/Library/Preferences/kicad/scripting/plugins/KiBuzzard/__init__.py", line 1, in <module>
    from . import KiBuzzard
  File "/Users/sabas/Library/Preferences/kicad/scripting/plugins/KiBuzzard/KiBuzzard/__init__.py", line 15, in <module>
    from .dialog.dialog import Dialog
ImportError: No module named dialog.dialog
```
<img width="797" alt="Captura de Pantalla 2021-01-17 a la(s) 14 54 48" src="https://user-images.githubusercontent.com/2874292/104855719-f4014600-58d3-11eb-81ca-037c6e176fef.png">

**Comment by VedantParanjape** [comment]
I don't think it works this way. 

Follow the install instructions in the README, I'll summarise them here for ya

1) check which of the folder exists? if none does, create the first one
```
    ~/Library/Application Support/kicad/scripting/plugins or on newer versions
    ~/Library/Preferences/kicad/scripting/plugins
```
2) navigate into the folder and run these command

```
git clone https://github.com/VedantParanjape/KiBuzzard.git --recursive
pip3 install -r requirements.txt --user
```

3) Restart KiCad, you'll see a icon on upper toolbar, use it to make footprints, enter this into the dialog box and press enter: `"<test>"`, you'll get a label at 0,0

**Comment by sabas1080** [comment]
I think my error es my python version 2.7.16, sorry, I will change

**Comment by sabas1080** [comment]
I have created `/Library/Application Support/kicad/scripting/plugins` and moved the folder, but "ImportError: No module named KiBuzzard"

"~/Library/Preferences/kicad/" not exist in my system

I have the directory "/Users/sabas/Library/Preferences/kicad/"

**Comment by VedantParanjape** [comment]
Sorry, this seems to be kicad issue. I am afraid I can't help as I don't own an apple device.

**Comment by sabas1080** [comment]
I have installed this plugin https://github.com/openscopeproject/InteractiveHtmlBom in "/Users/sabas/Library/Preferences/kicad/" and work, Now that I know the path is ok, I will check

**Comment by dcelectr** [comment]
I partially got it working on Win10 KiCad ver. 5.1.9 by using your version and overwriting the files I found here: 
https://github.com/arturo182/KiBuzzard/commit/621c7ee9247106f22fd7e06f87db86707086a906
but when I hit Enter on the menu (after entering my text) nothing happens.
![KiBuzzard](https://user-images.githubusercontent.com/48794887/105062039-2a5fd200-5a48-11eb-9e81-0fc01dd71a06.png)

**Comment by VedantParanjape** [comment]
> I partially got it working on Win10 KiCad ver. 5.1.9 by using your version and overwriting the files I found here:
> [arturo182@621c7ee](https://github.com/arturo182/KiBuzzard/commit/621c7ee9247106f22fd7e06f87db86707086a906)
> but when I hit Enter on the menu (after entering my text) nothing happens.
> ![KiBuzzard](https://user-images.githubusercontent.com/48794887/105062039-2a5fd200-5a48-11eb-9e81-0fc01dd71a06.png)

Because his changes are for KiCad 5.99, if you do the same changes on mine it should work. It appears you have done so, check for the label on 0,0

**Comment by dcelectr** [comment]
@VedantParanjape Yes, his version is for KiCad 5.99, but your version does not even appear on the bar with the plugins (I have Win10). I saw how you handled the Version 5.1.9, and how he implemented the Win10 compatibility. I tried to cherrypick his changes that made it work on Win10 and Implemented them in your version, but something I am not doing right, and I get error messages. The label does not appear on 0,0.

**Comment by VedantParanjape** [comment]
> @VedantParanjape Yes, his version is for KiCad 5.99, but your version does not even appear on the bar with the plugins (I have Win10). 

What does this even mean, check the issue referenced by https://github.com/VedantParanjape/KiBuzzard/issues/3. I guess you didn't even clone the submodule.

Please move further discussion there.

**Comment by omzlo** [comment]
I have the same issue, also on a Mac with KiCad 5.1.9.
I have the directory `~/Library/Preferences/kicad/scripting/plugins/`, which already contains another plugin called `InteractiveHtmlBom`. 
I did: `git clone https://github.com/VedantParanjape/KiBuzzard.git --recursive`:
```
Cloning into 'KiBuzzard'...
remote: Enumerating objects: 93, done.
remote: Counting objects: 100% (93/93), done.
remote: Compressing objects: 100% (56/56), done.
remote: Total 93 (delta 38), reused 84 (delta 29), pack-reused 0
Unpacking objects: 100% (93/93), done.
Submodule 'deps/Buzzard' (https://github.com/VedantParanjape/Buzzard.git) registered for path 'deps/buzzard'
Cloning into '/Users/xxxxxxxxxxxx/Library/Preferences/kicad/scripting/plugins/KiBuzzard/deps/buzzard'...
remote: Enumerating objects: 3, done.        
remote: Counting objects: 100% (3/3), done.        
remote: Compressing objects: 100% (3/3), done.        
remote: Total 709 (delta 0), reused 1 (delta 0), pack-reused 706        
Receiving objects: 100% (709/709), 4.58 MiB | 5.23 MiB/s, done.
Resolving deltas: 100% (338/338), done.
Submodule path 'deps/buzzard': checked out '17aeec6f2f48a71701409e72da172b584bbc3185'
```
I followed that with `pip3 install -r requirements.txt --user` which was successful.

Now the plugin directory contains two subdirectories `InteractiveHtmlBom` and the new `KiBuzzard`.

When I launch KiCad, there is nothing related to KiBuzzard in the PcbNew taskbar.

Out of curiosity, I opened the python shell, and typed `import KiBuzzard` and got the following error message:
```
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/Users/xxxxxxxxxxxx/Library/Preferences/kicad/scripting/plugins/KiBuzzard/__init__.py", line 1, in <module>
    from . import KiBuzzard
  File "/Users/xxxxxxxxxxxx/Library/Preferences/kicad/scripting/plugins/KiBuzzard/KiBuzzard/__init__.py", line 15, in <module>
    from .dialog.dialog import Dialog
ImportError: No module named dialog.dialog
```

**Comment by VedantParanjape** [comment]
@omzlo It seems to be something mac specific, some special way to import modules in mac, try changing `from .dialog.dialog import Dialog` to ` from dialog import Dialog` 

I think, it's a python issue, what's your python version ?

**Comment by omzlo** [comment]
@VedantParanjape I found a way to remove the dialog import issue: symply create an empty `KiBuzzard/KiBuzzard/dialog/__init__.py` file.

Once that is done, the icon of the plugin will show up in pcbnew. However, clicking on the icon shows an error message:
```
Traceback (most recent call last):

  File "/Users/xxxxxxxxxxx/Library/Preferences/kicad/scripting/plugins/KiBuzzard/KiBuzzard/__init__.py", line 158, in Run
    dlg = Dialog(self._pcbnew_frame, self.last_str, run_buzzard)

  File "/Users/xxxxxxxxxxx/Library/Preferences/kicad/scripting/plugins/KiBuzzard/KiBuzzard/dialog/dialog.py", line 11, in __init__
    dialog_base.KiBuzzardDialog.__init__(self, parent)

  File "/Users/xxxxxxxxxxx/Library/Preferences/kicad/scripting/plugins/KiBuzzard/KiBuzzard/dialog/dialog_base.py", line 22, in __init__
    self.SetSizeHints( wx.DefaultSize, wx.DefaultSize )

  File "/private/var/folders/r8/0nftyw1j4lq992n9q7dk2ztm0000gn/T/AppTranslocation/7165E78C-457B-4A8C-A1C3-506F289FB34F/d/kicad.app/Contents/Frameworks/python/site-packages/wx-3.0-osx_cocoa/wx/_core.py", line 9718, in SetSizeHints
    return _core_.Window_SetSizeHints(*args, **kwargs)

TypeError: in method 'Window_SetSizeHints', expected argument 2 of type 'int'
```

But I guess this progress. For your information, here's my python info:
```
py 0.9.8
Python 2.7.16 (default, Dec 23 2020, 16:11:24) 
[GCC 4.2.1 Compatible Apple LLVM 10.0.1 (clang-1001.0.46.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
Startup script executed: /Users/xxxxxxxxxxx/Library/Preferences/kicad/PyShell_pcbnew_startup.py
```

**Comment by VedantParanjape** [comment]
> @VedantParanjape I found a way to remove the dialog import issue: symply create an empty `KiBuzzard/KiBuzzard/dialog/__init__.py` file.
> 
> Once that is done, the icon of the plugin will show up in pcbnew. However, clicking on the icon shows an error message:
> ```
> Traceback (most recent call last):
> 
>   File "/Users/xxxxxxxxxxx/Library/Preferences/kicad/scripting/plugins/KiBuzzard/KiBuzzard/__init__.py", line 158, in Run
>     dlg = Dialog(self._pcbnew_frame, self.last_str, run_buzzard)
> 
>   File "/Users/xxxxxxxxxxx/Library/Preferences/kicad/scripting/plugins/KiBuzzard/KiBuzzard/dialog/dialog.py", line 11, in __init__
>     dialog_base.KiBuzzardDialog.__init__(self, parent)
> 
>   File "/Users/xxxxxxxxxxx/Library/Preferences/kicad/scripting/plugins/KiBuzzard/KiBuzzard/dialog/dialog_base.py", line 22, in __init__
>     self.SetSizeHints( wx.DefaultSize, wx.DefaultSize )
> 
>   File "/private/var/folders/r8/0nftyw1j4lq992n9q7dk2ztm0000gn/T/AppTranslocation/7165E78C-457B-4A8C-A1C3-506F289FB34F/d/kicad.app/Contents/Frameworks/python/site-packages/wx-3.0-osx_cocoa/wx/_core.py", line 9718, in SetSizeHints
>     return _core_.Window_SetSizeHints(*args, **kwargs)
> 
> TypeError: in method 'Window_SetSizeHints', expected argument 2 of type 'int'
> ```
> 
> But I guess this progress. For your information, here's my python info:
> ```
> py 0.9.8
> Python 2.7.16 (default, Dec 23 2020, 16:11:24) 
> [GCC 4.2.1 Compatible Apple LLVM 10.0.1 (clang-1001.0.46.4)] on darwin
> Type "help", "copyright", "credits" or "license" for more information.
> Startup script executed: /Users/xxxxxxxxxxx/Library/Preferences/kicad/PyShell_pcbnew_startup.py
> ```

ahhhh, I see what's the issue. upgrade to python3, I mean default python should be python3

I'll link this issue, see how it was solved.

https://github.com/VedantParanjape/KiBuzzard/issues/1#issuecomment-762264329

@sabas1080

**Comment by omzlo** [comment]
@VedantParanjape OK this makes sense. Unfortunately, on MacOS, KiCad comes with it's own version of python, which is hard-linked. I guess using Python 3 on MacOS means recompiling the app myself, which is probably not worth the effort/time.

**Comment by VedantParanjape** [comment]
@omzlo https://github.com/VedantParanjape/KiBuzzard/issues/3#issuecomment-770572193 check this comment

**Comment by CaptClaude** [comment]
I don't think your issues are due to the native version of Python (2.x).  The plugin explicitly calls Python3 in the code to run Buzzard for both Windows and Linux/MacOS:
```
if sys.platform.startswith('linux') or sys.platform == 'darwin':
    process = subprocess.Popen(['python3', buzzard_script] + args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
else:
   process = subprocess.Popen(['python3', buzzard_script] + args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
```

I am trying to get KiBuzzard to work on MacOS 10.14 with KiCAD 5.16.  After installing Python 3.7.9, I was changed to `/Users/jm/Library/Preferences/kicad` and created the additional two directories `/scripting/plugins`. From `plugins` I ran 
>$ git clone https://github.com/VedantParanjape/KiBuzzard.git --recursive
>$ pip3 install -r requirements.txt --user

No errors. Started KiCAD, refreshed plugins and there was a button on the toolbar.
Clicking the button brings up a dialog box.
<img width="438" alt="KiBuzzard-Dialog 2" src="https://user-images.githubusercontent.com/19495272/110064826-7edaba80-7d33-11eb-8d56-ff086d6974b2.png">
Clicking on the Create! button freezes KiCAD until I close the dialog box.  The dialog allows me to enter the label string manually but pressing enter gives the same results.  No label is created at (0,0).
Bottom line: It is possible to install KiBuzzard and it runs, but not to completion.  I've tried a few things in `__init__.py` but clearly my Python skills are not up to debugging.  I can run Buzzard from a command line and it works and outputs what appears to be a label footprint file, but I don't know where to go from there.  I will keep trying.

**Comment by VedantParanjape** [comment]
@CaptClaude Run kicad through command line, you can then actually view the errors due to which it failed.

**Comment by CaptClaude** [comment]
"open kicad.app" launches the app in OS X but clearly isn't what it is you are recommending I do. Would you mind giving me a little more direction on how to do that? Google comes up empty on that.

**Comment by VedantParanjape** [comment]
> "open kicad.app" launches the app in OS X but clearly isn't what it is you are recommending I do. Would you mind giving me a little more direction on how to do that? Google comes up empty on that.

if you run kicad from terminal, and then use the plugin, error generated by inbuilt python interpreter will be shown in terminal

**Comment by CaptClaude** [comment]
You are correct, sort of.  Nothing showed in the terminal window, but I now know that there is a Python version crash happening.  Now I need to figure out how to make python = python3 under MacOS.

<img width="517" alt="KiBuzzardPythonVersionCrash" src="https://user-images.githubusercontent.com/19495272/110216189-4b9f4500-7e73-11eb-9ec8-245583a2f504.png">

**Comment by VedantParanjape** [comment]
> You are correct, sort of. Nothing showed in the terminal window, but I now know that there is a Python version crash happening. Now I need to figure out how to make python = python3 under MacOS.
> 
> Although the plugin specifically calls Python3 when it spawns Buzzard, tells me that there is a version of Python 2.7 embedded deep inside KiCAD -- something I did not know.
> 
> <img alt="KiBuzzardPythonVersionCrash" width="517" src="https://user-images.githubusercontent.com/19495272/110216189-4b9f4500-7e73-11eb-9ec8-245583a2f504.png">

yeah, macos and windows is a real pain to work with. Someone got it working on Windows and sent a PR, if you have a breakthrough let me know

**Comment by VedantParanjape** [comment]
This is not in active development and changes have been merged into upstream, go to upstream repo. Closing the issue.
